extensions/amazon-aws/src/glue.tsx

```diff
@@ -0,0 +1,291 @@
+async function RunGlueJob(glueJobName: string, mutate: MutatePromise<GlueJobRun[] | undefined>) {
+  mutate(new GlueClient({}).send(new StartJobRunCommand({ JobName: glueJobName })), {
+    optimisticUpdate(data) {
+      return data;
+    },
+  })
+    .then((response) => {
+      showToast({ style: Toast.Style.Success, title: "Triggered Glue job", message: "JobRunId: " + response.JobRunId });
+    })
+    .catch((error) => {
+      showFailureToast(error, { title: "Unable to trigger Glue Job" });
+    });
+}
+
+function GlueJobDefinition({ glueJobName: glueJobName }: { glueJobName: string }) {
+  const { data: glueJobDetails, isLoading } = useCachedPromise(fetchJobDetails, [glueJobName]);
+  const description = glueJobDetails?.Job?.Description
+    ? glueJobDetails?.Job?.Description!.toString()
+    : `No Descrtiption`;
+  return (
+    <Detail
+      navigationTitle="Glue Job Definition"
+      markdown={
+        `# ` +
+        glueJobName +
+        `\n\n` +
+        description +
+        `\n\n` +
+        `___\n## Job Definition\n` +
+        `\`\`\`json\n` +
+        JSON.stringify(glueJobDetails?.Job, null, 2) +
+        `\`\`\``
+      }
+      isLoading={isLoading}
+      actions={
+        <ActionPanel title="title">
+          <AwsAction.Console url={resourceToConsoleLink(glueJobDetails?.Job?.Name, "AWS::Glue::JobRuns")} />
+          <Action.CopyToClipboard content={JSON.stringify(glueJobDetails?.Job?.Name)} title="Copy Job Name" />
+          <Action.CopyToClipboard
+            content={JSON.stringify(glueJobDetails?.Job?.DefaultArguments)}
+            title="Copy Default Arguments"
+          />
+          <Action.CopyToClipboard content={JSON.stringify(glueJobDetails?.Job)} title="Copy Job Definition" />
+        </ActionPanel>
+      }
+      metadata={
+        <Detail.Metadata>
+          <Detail.Metadata.Label title="Role" text={glueJobDetails?.Job?.Role} />
+          <Detail.Metadata.Label title="LastModifiedOn" text={glueJobDetails?.Job?.LastModifiedOn?.toISOString()} />
+          <Detail.Metadata.Label title="WorkerType" text={glueJobDetails?.Job?.WorkerType} />
+          <Detail.Metadata.Label
+            title="MaxConcurrentRuns"
+            text={glueJobDetails?.Job?.ExecutionProperty?.MaxConcurrentRuns?.toString()}
+          />
+          <Detail.Metadata.Label title="MaxCapacity" text={glueJobDetails?.Job?.MaxCapacity?.toString()} />
+          <Detail.Metadata.Label title="NumerOfWorkers" text={glueJobDetails?.Job?.NumberOfWorkers?.toString()} />
+          <Detail.Metadata.Label title="Timeout" text={glueJobDetails?.Job?.Timeout?.toString()} />
+          <Detail.Metadata.Separator />
+          <Detail.Metadata.Label title="CreatedOn" text={glueJobDetails?.Job?.CreatedOn?.toISOString()} />
+          <Detail.Metadata.Label title="GlueVersion" text={glueJobDetails?.Job?.GlueVersion?.toString()} />
+          <Detail.Metadata.Label title="PythonVersion" text={glueJobDetails?.Job?.Command?.PythonVersion} />
```

Comment:
Same as the other comment. If some properties can be optional, we will need to render the `Detail.Metadata.Label` optionally as well.